### ISSUE TITLE
Is there a way to ask TildeArray to print all the intermediate quantities and operations (i.e. permutation, GEMM, etc.)?

### ISSUE THREAD

**Issue by xlliu98** [ValeevGroup/tiledarray#221]

**Comment by justusc** [comment]
Yes there is. You can find the interface code here. https://github.com/ValeevGroup/tiledarray/blob/master/src/TiledArray/expressions/expr_trace.h

It has been a while, but if I remember the code was
```c++
std::cout << c("i,k") << a("i,j") * b("i,k");
```
which prints out the equivalent to the expression
```c++
c("i,k") = a("i,j") * b("i,k");
```
the print does not actually evaluate the expression. It just walks the expression tree to generate the expected operations, including permutations.

It was not a feature that was often used, so I am not sure if the feature was broken or not implemented for certain expression types since I last worked on the code. If their are missing implementations, extending the code should be fairly simple as you would only need to add a print function that matches the interface of other expressions.

**Comment by xlliu98** [comment]
Thanks. Is there a similar way for block expressions?

**Comment by justusc** [comment]
I think that may not have been implemented.
